aws/aws_secrets_manager/aws_scm.go

```diff
@@ -175,15 +185,17 @@ func (a *AWSSecretsMgr) Rencrypt(
 }
 
 func (a *AWSSecretsMgr) get(secretID string) (map[string]interface{}, secrets.Version, error) {
-	secretValueOutput, err := a.scm.GetSecretValue(&secretsmanager.GetSecretValueInput{
+	secretValueOutput, err := a.scm.GetSecretValue(context.TODO(), &secretsmanager.GetSecretValueInput{
 		SecretId: aws.String(secretID),
 	})
 	if err != nil {
-		if aerr, ok := err.(awserr.Error); ok {
-			if aerr.Code() == secretsmanager.ErrCodeResourceNotFoundException {
+		var apiErr smithy.APIError
+		if errors.As(err, &apiErr) {
+			// aerr, ok := err.(awserr.Error); ok {
+			if apiErr.ErrorCode() == "ResourceNotFoundException" {
 				return nil, secrets.NoVersion, secrets.ErrInvalidSecretId
-			} else if aerr.Code() == secretsmanager.ErrCodeInvalidRequestException &&
-				strings.Contains(aerr.Error(), "marked for deletion") {
+			} else if apiErr.ErrorCode() == "InvalidRequestException" &&
```

Comment:
define these strings as constants

Reply:
Fixed in this [commit](https://github.com/libopenstorage/secrets/pull/76/commits/5f953e754f247b280ae7635029629ac2020fb1e3)